test/serializers/read_attribute_for_serialization_test.rb

```diff
@@ -7,20 +7,28 @@ class ReadAttributeForSerializationTest < ActiveSupport::TestCase
       class Parent < ActiveModelSerializers::Model
         attr_accessor :id
       end
+
       class Child < Parent
         attr_accessor :name
       end
+
       class ParentSerializer < ActiveModel::Serializer
         attributes :$id
 
         define_method(:$id) do
           object.id
         end
       end
+
       class ChildSerializer < ParentSerializer
         attributes :name
       end
 
+      def test_for_serializer_initialized_with_nil_object
+        child = ChildSerializer.new(nil)
+        assert_equal nil, child.read_attribute_for_serialization(:name)
```

Comment:
For my particular case, I was using it like so: 
```EmployeeSerializer.new(employee.version_at(date)).attributes```
I half expected that if the passed in object is nil, I would be returned a nil or {} or something similar. Calling further methods on nil or {}, etc should raise exceptions in my own code that I am very familiar with and I won't need to look at the gem's code.

Another scenario off the top of my head would be something like 
```EmployeeSerializer.new(get_employee_for_serializing).attributes```

where ```get_employee_for_serializing``` may be a method that may return a nil when you do something like a #find or Model.where(conditions).first. I can see such helper methods being used in controllers rendering json via serializers.

I realize that adding a nil check would be very easy in user's code. But IMHO I should put these in after realizing that I cannot pass in nil to a serializer after seeing a TypeError.